Production/test/unitTest.py

```diff
@@ -1,92 +1,157 @@
+        self.command = command
+        self.redir = redir
+        self.cmd = "cmsRun"
+        self.config = "runMakeTreeFromMiniAOD_cfg.py"
+        self.array = self.getArray()
+
+    def getArray(self):
+        mytest = []
+        mytest += [self.cmd, self.config, "scenario="+self.scenario]
+        # different methods for input files
+        if len(self.dataset)>0:
+            mytest.append("dataset="+self.dataset)
+        else:
+            mytest.append("inputFilesConfig="+self.inputFilesConfig)
+            mytest.append("nstart="+str(self.nstart))
+            mytest.append("nfiles="+str(self.nfiles))
+        # different shell commands
+        mytest += ["outfile="+self.outfile, "numevents="+str(self.numevents)]
+        if len(self.command)>0: mytest.append(self.command)
+        if len(self.redir)>0: mytest.append("redir="+self.redir)
+        mytest.append(self.getLogName())
+        return mytest
+
+    def getLogName(self):
+        return self.outfile+".log"
+
+    def getROOTName(self):
+        return self.outfile+"_RA2AnalysisTree.root"
+
+    def printTest(self, itest, log):
+        logname = self.array[-1]
+        tmp = str(itest)+":\n  "+" \\\n  ".join(self.array[0:-1])
+        if log:
+            tmp +=" \\\n "+" >& "+logname+" &"
+        print tmp
+
+    def forkAndRun(self, clean):
         # fork the cmsRun process and exit
         fork_pid = os.fork()
         if fork_pid == 0:
-            with open(mytests[test][-1],'w') as out:
-                p = subprocess.Popen(mytests[test][0:-1], stdin=None, stdout=out, stderr=out, close_fds=True)
+            with open(self.getLogName(),'w') as out:
+                p = subprocess.Popen(self.array[0:-1], stdin=None, stdout=out, stderr=out, close_fds=True)
                 print "\nRunning test... ["+str(p.pid)+"]"
                 sts = os.waitpid(p.pid, 0)[1]
                 print "\nTest is done! ["+str(p.pid)+"]"
+                if clean: self.cleanFiles(clean)
+
+    def run(self, log, clean):
+        out = None
+        if log: out = open(self.getLogName(),'w')
+        p = subprocess.Popen(self.array[0:-1], stdin=None, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, close_fds=True, bufsize=1)
+        print "\nRunning test..."
+        with p.stdout:
+            for line in iter(p.stdout.readline, b''):
+                print line,
+                if log: out.write(line)
+        p.wait()
+        if log: out.close()
+        print "\nTest is done!"
+        if clean: self.cleanFiles(clean)
+
+    def cleanFiles(self, clean):
+        print "\nCleaning output files..."
+        files=[self.getROOTName(),self.getLogName()]
+        if clean>0: files=files[0:clean]
+        for file in files:
+            if os.path.exists(file):
+                os.remove(file)
+        print "\nDone cleaning files!"
+
+def defineTests(mytests, scenario, name, numevents, command, dataset, inputFilesConfig):
+    # activate these when the requisite MC exists
```

Comment:
this comment can probably be deleted, it's very outdated

Reply:
Done